lib/utils/fncache_test.go

```diff
@@ -178,14 +181,21 @@ func TestFnCacheCancellation(t *testing.T) {
 	// unblock the loading operation which is still in progress
 	close(blocker)
 
-	ctx, cancel = context.WithTimeout(context.Background(), timeout)
+	// use long timeout since we need to wait for some concurrent operations to complete
+	// behind the scenes, and that can take a while in a resource-constrained environment.
+	ctx, cancel = context.WithTimeout(context.Background(), longTimeout)
 	defer cancel()
 
+	loadFnWasRun := atomic.NewBool(false)
```

Comment:
I'm not 100% sure, but I think this will panic on a 32-bit arch. We've already had issues like that https://github.com/gravitational/teleport/pull/11822

Reply:
Can we just use stdlib atomic int64 for now? Next version of Go will have a built-in atomic bool.

Reply:
> go.uber.org/atomic automatically handles alignment and protects from non-atomic access.

I don't think this is entirely true, ex: https://github.com/uber-go/atomic/issues/105
But I checked, and this use seems to work on the 32bit arch. I didn't realize that this is `uber` atomic, not `stdlib`.
I agree with @zmb3 that we should prefer stuff that is in the stdlib, but atomics is one of those things that so far causes us a lot of problems, and Go 1.19 is not available for us now.

@fspmarshall Do you know why did we decide to stop using Uber atomics?

Reply:
I'd consider the linked issue to be an abuse of the API rather than a failure of the library.  They are manually dereferencing the wrapper returned by `NewDuration()`, which breaks alignment.  IMO if you're dereferencing opaque types returned by concurrency libraries, you should expect some problems 😬 

> Do you know why did we decide to stop using Uber atomics?

If we stopped, I certainly wasn't aware of it.  I always encourage people to use it when I'm reviewing, and I use it in all the code I write 🤷 The team was a lot smaller [back then](https://github.com/gravitational/teleport/pull/3536#discussion_r406474964).  Probably just something that got lost along the way.

We do deliberately avoid it in the `api` package because we are a more aggressive about introducing external deps in there.

Reply:
I think that we/I just lost the context. When I was looking at our code, I wasn't aware that we were using Uber atomics as most of the usage that I saw used the `std` library.